data/transactions/signedtxn.go

```diff
@@ -59,9 +61,18 @@ type SignedTxnWithAD struct {
 
 // ID returns the Txid (i.e., hash) of the underlying transaction.
 func (s SignedTxn) ID() Txid {
+	if s.cachedID != nil {
```

Comment:
Curious: did we test performance both with and without the `cachedID` change?

Reply:
I do not recall but can perf tests

Reply:
w/o cache id change the gain is +400 TPS